src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +346,322 @@ void func_80B67540(EnSth* this, PlayState* play) {
         case TEXT_STATE_5:
             if (Message_ShouldAdvance(play)) {
                 switch (play->msgCtx.currentTextId) {
-                    case 0x1134:
+                    case 0x1134: // (does not exist)
                         func_80151938(play, play->msgCtx.currentTextId + 1);
                         break;
 
-                    case 0x1132:
-                    case 0x113A:
-                    case 0x113F:
-                        func_80151938(play, 0x1133);
+                    case 0x1132:                     // Heard noise, came in to see
+                    case 0x113A:                     // Had no idea there was a basement here
+                    case 0x113F:                     // Heard noise... I never thought I'd find a place like this
+                        func_80151938(play, 0x1133); // did you find this place?
                         break;
 
-                    case 0x1133:
-                        func_80151938(play, 0x1136);
-                        func_80B670A4(this, 6);
+                    case 0x1133:                     // did you find this place?
+                        func_80151938(play, 0x1136); // I want to buy this place from you
+                        EnSth_SwitchAnimation(this, STH_ANIM_PLEAD);
                         break;
 
-                    case 0x1136:
+                    case 0x1136: // I want to buy this house from you
                         SET_WEEKEVENTREG(WEEKEVENTREG_13_80);
 
-                        switch (sp2C) {
-                            case 0:
-                                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_13_40)) {
-                                    this->actor.home.rot.z = 6;
+                        switch (day) {
+                            case 0: // first day
+                                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_GIANTS_WALLET)) {
+                                    STH_GI_ID(&this->actor) = GI_RUPEE_SILVER;
                                 } else {
-                                    SET_WEEKEVENTREG(WEEKEVENTREG_13_40);
+                                    SET_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_GIANTS_WALLET);
                                     switch (CUR_UPG_VALUE(UPG_WALLET)) {
                                         case 0:
-                                            this->actor.home.rot.z = 8;
+                                            STH_GI_ID(&this->actor) = GI_WALLET_ADULT;
                                             break;
```

Comment:
If the Oceanside spiderhouse is your first wallet upgrade, then you'd get `GI_WALLET_ADULT` right? So then why would `WEEKEVENTREG_RECEIVED_GIANTS_WALLET` be set? Is this flag more suppose to be `WEEKEVENTREG_RECEIVED_OCEANSIDE_WALLET_UPGRADE`?

Reply:
That's a fair take, I don't know how anyone would get this wallet first unless they somehow completely missed the bank, but this is more accurate.